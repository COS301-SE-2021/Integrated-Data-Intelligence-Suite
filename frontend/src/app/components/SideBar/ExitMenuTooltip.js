import React, {Component} from "react";
import {Menu} from "antd";
import {LockOutlined, PoweroffOutlined} from "@ant-design/icons";
import {
    BrowserRouter as Router,
    Switch,
    Route, Link
} from "react-router-dom";
import LoginPage from "../../pages/LoginPage";
import GraphList from "../ContentSection/GraphList";
import Register from "../../pages/Register";

// import {Link} from "react-router-dom";

class ExitMenuTooltip extends React.Component {
    render() {
        return (
            <>
                <Menu id={'exit_menu'}>
                    <Menu.Item key="1" icon={<LockOutlined/>}>
                        Lock
                    </Menu.Item>
                    <Menu.Item key="2" icon={<PoweroffOutlined/>}>
                                <Link to="/login">logout</Link>
                    </Menu.Item>
                </Menu>
            </>
        );
    }
}

export default ExitMenuTooltip;