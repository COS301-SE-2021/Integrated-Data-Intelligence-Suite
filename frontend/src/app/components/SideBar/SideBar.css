/*************************************************************/
/********************* Sidebar(Left)  ***************************/
#sidebar_div[style] {
    background: white;
    border-right: 0.2em solid #E8E8E9;
    /*border: 3px solid #5873f9;*/
    /*width: 168px !important;*/
    /*min-width: 0 !important;*/
    /*max-width: 190px !important;*/
    width: 160px !important;
    min-width: 0 !important;
    max-width: 160px !important;
}

#sidebar_menu {
    background: transparent;
    color: #333333;
    border: 0;
    /*border: 3px solid red;*/
    height: 87%;
}

#sidebar_menu > li {
    padding-left: 24px;
    /*border: 2px solid red;*/
    width: 100%;
    /*margin-left: 12px;*/
    margin-right: 0px;
    margin-bottom: 14px;
    margin-top: 8px;
    padding-right: 24px;
    /*border-radius: 6px;*/
    padding-bottom: 11px;
    padding-top: 10px;
    height: 45px;
    background: transparent;
}

/*When a Sidebar Menu item when it has been selected(clicked)*/
#sidebar_menu > li.ant-menu-item-selected {
    border: 0;
    /*background: #5773FA;*/
    color: var(--neon-blue);
    border-right: 3px solid #5773FA;
    background: white;

}

/*When a Sidebar Menu item when it has been Hovered over*/
#sidebar_menu > li.ant-menu-item-active {
    /*border: 0;*/
    /*background: #5773FA;*/
    color: #5773FA;
    border-right: 3px solid #5773FA;
}

/*Hovering over a menu item that has been selected*/
#sidebar_menu > li.ant-menu-item-active.ant-menu-item-selected {
    /*border: 0;*/
    /*color: white;*/
}

/**/
#sidebar_menu > li.ant-menu-item::after {
    border: 0;
    color: white;
}


/* The Collapse/Expand Trigger(arrow) at the Top of the SideBar*/
.ant-layout-sider-trigger[style] {
    /* Positions the Trigger at the top of*/
    /*Comment this out and Set 'bottom:0' to position trigger at the bottom*/
    bottom: 0;
    /*max-width: 168px !important;*/
    /*width: 168px !important;*/
    /*margin: 0;*/
    /*max-width: 190px !important;*/
    background: white;
    color: black;
    width: 80px !important;
    min-width: 0 !important;
    max-width: 160px !important;
}

#outer_layout {
    min-height: 100vh;
    background: black;
}

#logo {
    height: 10%;
    /*margin: 16px;*/
    margin: 0;
    /*background: rgb(252, 0, 64);*/
    /*line-height: 3px;*/
    /*border: 3px solid black;*/
    /*display: grid;*/
    /*grid-template-columns: 100%;*/
    /*grid-template-rows: 100%;*/
    background: white;
    /*background-image: url("../../assets/athena-logo.png") !important;*/
    background-size: cover;
    background-position: center;
    color:black;
    /*align-content: center;*/
}

#logo-image[style] {
    background-size: cover;
}


.ant-layout-sider {
    color: #fff;
    line-height: 120px;
    /*background: #3ba0e9;*/
}

> .code-box-demo > .ant-layout + .ant-layout {
    margin-top: 48px;
}

/*Color of a Menu Item When Selected*/
/*.ant-menu-dark.ant-menu-dark:not(.ant-menu-horizontal) .ant-menu-item-selected {*/
/*    background-color: deeppink;*/
/*}*/

[data-theme="dark"] .ant-layout-sider {
    background: #5773FA;
}

