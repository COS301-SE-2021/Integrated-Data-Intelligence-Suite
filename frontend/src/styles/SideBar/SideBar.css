/*************************************************************/
/********************* Sidebar(Left)  ***************************/
#sidebar_div {
    background: #95DBB6;
}

#sidebar_menu {
    background: transparent;
    color: white;
    border: 0;
    /*border: 3px solid red;*/
    height:87%;
}

#sidebar_menu > li {
    padding-left: 24px;
    /*border: 2px solid red;*/
    width: 80%;
    margin-left: 12px;
    margin-right: 0px;
    margin-bottom: 14px;
    margin-top: 8px;
    padding-right: 24px;
    border-radius: 6px;
    padding-bottom: 11px;
    padding-top: 10px;
    height: 45px;
    background: transparent;
}

/*When a Sidebar Menu item when it has been selected(clicked)*/
#sidebar_menu > li.ant-menu-item-selected {
    border: 0;
    background: #15B761;
    color: white;
}

/*When a Sidebar Menu item when it has been Hovered over*/
#sidebar_menu > li.ant-menu-item-active {
    border: 0;
    color: #15B761;
}

/*Hovering over a menu item that has been selected*/
#sidebar_menu > li.ant-menu-item-active.ant-menu-item-selected {
    border: 0;
    color: white;
}


/**/
#sidebar_menu > li.ant-menu-item::after {
    border: 0;
    color: white;
}


/* The Collapse/Expand Trigger(arrow) at the Top of the SideBar*/
.ant-layout-sider-trigger[style] {
    /* Positions the Trigger at the top of*/
    /*Comment this out and Set 'bottom:0' to position trigger at the bottom*/
    top: 0;
    position: fixed;
    z-index: 1;
    height: 48px;
    color: black !important;
    line-height: 48px;
    text-align: center;
    background:  white !important  ;
    cursor: pointer;
    transition: all 0.2s;
}

#outer_layout {
    min-height: 100vh;
    background: black;
}

#logo {
    height: 10%;
    /*margin: 16px;*/
    margin: 0;
    /*background: rgb(252, 0, 64);*/
    background: transparent;
    line-height: 3px;
    /*border: 3px solid #f44336;*/
}


.ant-layout-sider {
    color: #fff;
    line-height: 120px;
    /*background: #3ba0e9;*/
}

> .code-box-demo > .ant-layout + .ant-layout {
    margin-top: 48px;
}

/*Color of a Menu Item When Selected*/
.ant-menu-dark.ant-menu-dark:not(.ant-menu-horizontal) .ant-menu-item-selected {
    background-color: deeppink;
}

[data-theme="dark"] .ant-layout-sider {
    background: #3499ec;
}

