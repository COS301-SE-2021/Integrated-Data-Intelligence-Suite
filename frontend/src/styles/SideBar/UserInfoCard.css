/*************************************************************/
/********************* User Avatar  Card ***************************/
/************Found in the Bottom Left corner********************/

/* The div that holds the entire user_avatar card*/
#user_avatar_card {
    /*bottom: -50vh;*/
    bottom:0;
    height: 10%;
    /* color: black; */
    /*background: #15B761;*/
    background: transparent;
    /*border: 0;*/
    border: 2px solid yellow;
    /*margin-left: 8px;*/
    /*margin-right: 8px;*/
    /*border-radius: 6px;*/

}

#user_avatar_card > .ant-card-body {
    padding: 0;
    border: 0;
    border-radius: 6px;
    background: transparent;
}

#user_avatar_pic {
    /*border: 2px solid black;*/
    color: white;
    background: orange;
    float: right;
    width: 39px;
    height: 39px;
    line-height: 1.5715;
    font-size: 16px;
    border-radius: 100%;
    margin-left: 4px;
}

#meta_id {
     border: 2px solid purple;
    /*border: 0;*/
    float: right;
    /*width: 78%;*/
    /*height: 100%;*/
    background: #04aa6d00;

}

.ant-card-meta {
    /*border: 2px solid blue;*/
    margin: 0;
    background: transparent;
    border: 0;

}

/*The div wrapping the users avatar*/
#meta_id > .ant-card-meta-avatar {
    float: left;
    padding-right: 0;
    padding-left: 0;
    margin-top: 18px;
    background: transparent;
    border: 0;
}

/*The div for the users Name and a description*/
#meta_id > .ant-card-meta-detail {
    float: left;
    justify-content: left;
    /*margin-top: 19px;*/
    background: transparent;
    border: 0;

}

/*The users' Name*/
#meta_id > .ant-card-meta-detail > .ant-card-meta-title {
    color: white;
    border: 0;
    font-size: 13px;
    font-weight: 600;
    margin-bottom: 0;
    margin-top: 25px;
    background: transparent;
    border: 0;

}

/*The grey out section underneath the name*/
#meta_id > .ant-card-meta-detail > .ant-card-meta-description {
    font-size: 11px;
    height: 17px;
    width: 12px;
    color: #9c9494;
    background: transparent;
}

/*The ellipsis button*/
#exit_menu_button {
    /*border: 1px solid green;*/
    border: 0;
    font-size: 0;
    float: right;
    /*transform: rotate(*/
    /*        90deg*/
    /*);*/
    margin-top: 24px;
    width: 31px;
    height: 23px;
    padding: 0;
    background: transparent;
}

#exit_menu_button > .exit_menu_ellipsis_icon > svg {
    color: white;
    background: transparent;
}


