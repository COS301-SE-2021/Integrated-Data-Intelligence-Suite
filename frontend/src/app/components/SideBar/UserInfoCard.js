import {Button, Card, Dropdown, Menu, Skeleton, Avatar } from "antd";
// import Avatar from "antd/es/avatar/avatar";
import {UpOutlined} from "@ant-design/icons";
import React, {Component} from 'react';
import ExitMenuTooltip from "./ExitMenuTooltip";
const { Meta } = Card;

function setUserName(){
    const localUser = localStorage.getItem("user");
    if(localUser){
        return JSON.parse(localUser);
    }else{
        return "user";
    }
}

class UserInfoCard extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            loading: true,
            collapsed: false,
            user: setUserName()
        };
    }


    onChange = checked => {
        this.setState({loading: !checked});

    };

    onCollapse(collapsed) {
        if (this.state.collapsed) {


        }

    }

    render() {
        const {loading, user} = this.state;

        return (
            <>
                {/*<Switch checked={!loading} onChange={this.onChange} style={{width: 40}}/>*/}
                <Card id={'user_avatar_card'}>
                    <Skeleton loading={false} avatar active>


                        {/*The drop down menu that allows the user to log out or lock the app*/}
                        <Dropdown
                            overlay={<ExitMenuTooltip/>}
                            placement="topCenter"
                            arrow={true}
                            trigger={'click'}
                        >
                            <Button
                                id={'exit_menu_button'}
                                icon={
                                    <UpOutlined
                                        className={'exit_menu_ellipsis_icon'}
                                    />
                                }
                            >
                            </Button>
                        </Dropdown>

                        {/*The user Avatar Image*/}
                        <Meta
                            id={'meta_id'}
                            className={'user_meta_card'}
                            title={user.username}
                            // description={'name@example.com'}
                            // avatar={
                            //     <Avatar
                            //         id={'user_avatar_pic'}
                            //         shape={'round'}
                            //     >
                            //         {user.username.substring(0,1)}
                            //     </Avatar>
                            // }
                        />
                    </Skeleton>
                </Card>
            </>
        );
    }
}

export default UserInfoCard;
