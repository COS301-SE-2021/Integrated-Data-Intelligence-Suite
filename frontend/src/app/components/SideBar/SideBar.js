import { Menu, Layout } from 'antd';
import { BarChartOutlined, HomeOutlined, SettingOutlined } from '@ant-design/icons';
import React, { Component } from 'react';
import { BrowserRouter, Link, Router } from 'react-router-dom';
import UserInfoCard from '../UserInfoCard/UserInfoCard';
import { AiOutlineHeart, AiOutlineHome } from 'react-icons/all';

const {
    Header,
    Footer,
    Sider,
    Content,
} = Layout;

class SideBar extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            collapsed: false,
            active: '2',
        };

        this.setActive = (value) => {
            this.setState({ active: value });
        };

        this.onCollapse = (collapsed) => {
            console.log(`IsCOllapsed: ${collapsed}`);
            this.setState({ collapsed });
        };
    }

    render() {
        const {
            collapsed,
            active
        } = this.state;
        console.log('active comp ', active);
        return (
            <>

                <Sider
                    collapsible
                    collapsed={collapsed}
                    onCollapse={this.onCollapse}
                    id="sidebar_div"
                >
                    <div id="logo">
                        <div id={'logo-image'}/>
                    </div>
                    <Menu
                        id="sidebar_menu"
                        theme="light"
                        defaultSelectedKeys={active || '2'}
                        mode="inline"
                    >

                        <Menu.Item
                            key="1"
                            icon={<AiOutlineHome/>}
                            onClick={() => this.setActive('1')}
                        >
                            Home
                            <Link to="/"/>
                        </Menu.Item>


                        <Menu.Item
                            key="2"
                            icon={<BarChartOutlined/>}
                            onClick={() => this.setActive('2')}
                        >
                            Analytics
                            <Link to="/"/>
                        </Menu.Item>

                        <Menu.Item
                            key="3"
                            icon={<SettingOutlined/>}
                            onClick={() => this.setActive('3')}
                        >
                            Settings
                            <Link to="/settings"/>
                        </Menu.Item>

                        <Menu.Item
                            key="4"
                            icon={<AiOutlineHeart/>}
                            onClick={() => this.setActive('4')}
                        >
                            Credits
                            <Link to="/"/>
                        </Menu.Item>

                    </Menu>
                </Sider>
            </>
        );
    }
}

export default SideBar;
