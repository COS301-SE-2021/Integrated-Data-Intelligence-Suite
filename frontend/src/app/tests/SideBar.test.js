import React from 'react';
import renderer from 'react-test-renderer';
import SideBar from '../components/SideBar/SideBar';
import {BrowserRouter} from "react-router-dom";
import {unmountComponentAtNode} from "react-dom";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});


it('renders correctly', () => {
    const tree = renderer
        .create(<BrowserRouter>

                <SideBar/>
            </BrowserRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});