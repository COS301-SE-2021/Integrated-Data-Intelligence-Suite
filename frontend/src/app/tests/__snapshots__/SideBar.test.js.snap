// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<aside
  className="ant-layout-sider ant-layout-sider-dark"
  id="sidebar_div"
  style={
    Object {
      "flex": "0 0 200px",
      "maxWidth": "200px",
      "minWidth": "200px",
      "width": "200px",
    }
  }
>
  <div
    className="ant-layout-sider-children"
  >
    <div
      id="logo"
    />
    <ul
      className="ant-menu ant-menu-root ant-menu-inline ant-menu-light"
      data-menu-list={true}
      id="sidebar_menu"
      onKeyDown={[Function]}
      role="menu"
      tabIndex={0}
    >
      <li
        className="ant-menu-item ant-menu-item-selected"
        data-menu-id="sidebar_menu-1"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        tabIndex={-1}
      >
        <span
          aria-label="home"
          className="anticon anticon-home ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="home"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
            />
          </svg>
        </span>
        <span
          className="ant-menu-title-content"
        >
          Home
          <a
            href="/"
            onClick={[Function]}
          />
        </span>
      </li>
      <li
        className="ant-menu-item"
        data-menu-id="sidebar_menu-2"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        tabIndex={-1}
      >
        <span
          aria-label="bar-chart"
          className="anticon anticon-bar-chart ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="bar-chart"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm-600-80h56c4.4 0 8-3.6 8-8V560c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v144c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V384c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v320c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V462c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v242c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V304c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v400c0 4.4 3.6 8 8 8z"
            />
          </svg>
        </span>
        <span
          className="ant-menu-title-content"
        >
          Charts
          <a
            href="/chart"
            onClick={[Function]}
          />
        </span>
      </li>
      <li
        className="ant-menu-item"
        data-menu-id="sidebar_menu-9"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        tabIndex={-1}
      >
        <span
          aria-label="setting"
          className="anticon anticon-setting ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="setting"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
            />
          </svg>
        </span>
        <span
          className="ant-menu-title-content"
        >
          Settings
          <a
            href="/settings"
            onClick={[Function]}
          />
        </span>
      </li>
    </ul>
    <div
      aria-hidden={true}
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
</aside>
`;
